lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -34,6 +37,70 @@ namespace circt {
 /// Calyx component.
 using FuncMapping = DenseMap<FuncOp, calyx::ComponentOp>;
 
+using Scheduleable = std::variant<calyx::GroupOp>;
+
+//===----------------------------------------------------------------------===//
+// Utility functions
+//===----------------------------------------------------------------------===//
+
+/// Tries to match a constant value defined by op. If the match was
+/// successful, returns true and binds the constant to 'value'.
```

Comment:
Can you describe any side effects of `value` if it returns false please? Or, if there are none, make that known.